packages/js/product-editor/src/blocks/generic/linked-product-list/edit.tsx

```diff
@@ -0,0 +1,255 @@
+		return function handleRemoveProductClick() {
+			const newLinkedProductIds = removeLinkedProductDispatcher(
+				product,
+				state.linkedProducts
+			);
+
+			setLinkedProductIds( newLinkedProductIds );
+		};
+	}
+
+	return (
+		<div { ...blockProps }>
+			<div className="wp-block-woocommerce-product-linked-list-field__form-group-content">
+				<SelectControl< Product >
+					{ ...selectProps }
+					items={ state.searchedProducts }
+					placeholder={ __( 'Search for products', 'woocommerce' ) }
+					label=""
+					selected={ null }
+					onSelect={ handleSelect }
+					__experimentalOpenMenuOnFocus
+				>
+					{ ( {
+						items,
+						isOpen,
+						highlightedIndex,
+						getMenuProps,
+						getItemProps,
+					} ) => (
+						<Menu
+							isOpen={ isOpen }
+							getMenuProps={ getMenuProps }
+							className="wp-block-woocommerce-product-linked-list-field__menu"
+						>
+							{ isFetching ? (
+								<div className="wp-block-woocommerce-product-linked-list-field__menu-loading">
+									<Spinner />
+								</div>
+							) : (
+								items.map( ( item, index ) => (
+									<MenuItem< Product >
+										key={ item.id }
+										index={ index }
+										isActive={ highlightedIndex === index }
+										item={ item }
+										getItemProps={ ( options ) => ( {
+											...getItemProps( options ),
+											className:
+												'wp-block-woocommerce-product-linked-list-field__menu-item',
+										} ) }
+									>
+										<>
+											<div
+												className="wp-block-woocommerce-product-linked-list-field__menu-item-image"
+												style={ getProductImageStyle(
+													item
+												) }
+											/>
+											<div className="wp-block-woocommerce-product-linked-list-field__menu-item-content">
+												<div className="wp-block-woocommerce-product-linked-list-field__menu-item-title">
+													{ item.name }
+												</div>
+
+												<FormattedPrice
+													product={ item }
+													className="wp-block-woocommerce-product-linked-list-field__menu-item-description"
+												/>
+											</div>
+										</>
+									</MenuItem>
+								) )
+							) }
+						</Menu>
+					) }
+				</SelectControl>
+			</div>
+
+			{ Boolean( state.linkedProducts.length ) && (
+				<div role="table">
+					<div role="rowgroup">
+						<div role="rowheader">
+							<div role="columnheader">
+								{ __( 'Product', 'woocommerce' ) }
+							</div>
+							<div
+								role="columnheader"
+								aria-label={ __( 'Actions', 'woocommerce' ) }
+							/>
+						</div>
+					</div>
+					<div role="rowgroup">
+						{ state.linkedProducts.map( ( product ) => (
+							<div role="row" key={ product.id }>
+								<div role="cell">
+									<div
+										className="wp-block-woocommerce-product-linked-list-field__product-image"
+										style={ getProductImageStyle(
+											product
+										) }
+									/>
```

Comment:
It would be nice to have this as a `ProductImage` component.

Reply:
done -> https://github.com/woocommerce/woocommerce/pull/43145/commits/3f922240b1c62c6e628c6db57477f53515742440